web-api/src/mind.rs

```diff
@@ -24,9 +27,107 @@ use rand::{
     thread_rng,
 };
 use serde::{de, Deserialize, Deserializer};
-use xayn_ai_coi::nan_safe_f32_cmp_desc;
+use xayn_ai_coi::{nan_safe_f32_cmp_desc, CoiConfig, CoiSystem};
+
+use crate::{
+    embedding::{self, Embedder},
+    models::{DocumentId, DocumentProperties, IngestedDocument, UserId, UserInteractionType},
+    personalization::{
+        routes::{
+            personalize_documents_by,
+            update_interactions,
+            PersonalizeBy,
+            UserInteractionData,
+        },
+        PersonalizationConfig,
+    },
+    storage::{self, memory::Storage},
+};
+
+struct State {
+    storage: Storage,
+    embedder: Embedder,
+    coi: CoiSystem,
+    personalization: PersonalizationConfig,
+}
+
+impl State {
+    fn new(storage: Storage) -> Result<Self, Error> {
+        let embedder = Embedder::load(&embedding::Config {
+            directory: "../assets/smbert_v0003".into(),
+            ..embedding::Config::default()
+        })?;
+        let coi = CoiConfig::default().build();
+        let personalization = PersonalizationConfig::default();
+
+        Ok(Self {
+            storage,
+            embedder,
+            coi,
+            personalization,
+        })
+    }
 
-use crate::models::DocumentId;
+    async fn insert(&self, documents: Vec<Document>) -> Result<(), Error> {
+        let documents = documents
+            .into_iter()
+            .map(|document| {
+                let snippet = if document.snippet.is_empty() {
+                    document.title.clone()
+                } else {
+                    document.snippet.clone()
+                };
```

Comment:
:question: In the ingestion, we use only this snippet. Any reason to use the title instead here?